ARE/middleware/src/main/java/eu/asterics/mw/utils/OSUtils.java

```diff
@@ -153,74 +154,94 @@ public static Process startApplication(String applicationPath, String arguments,
-                return null;
+            //if the cmdEndIndex < 0, we did not find a space or endQuote, so treat the whole string as command.
+            if (cmdEndIndex < 0) {
+                logger.warning("Only command, no arguments: " + applicationPathAndArguments);
+                cmdEndIndex=applicationPathAndArguments.length();
             }
+            // cmdEndIndex++;
+            String cmdString = applicationPathAndArguments.substring(0, cmdEndIndex);
+            // The command can be an absolute path, so ensure that it has proper system separators (\ or \\ or /)
+            command.add(FilenameUtils.separatorsToSystem(cmdString));
 
+            String arguments = "";
             try {
-                List<String> command = new ArrayList<String>();
-
-                int cmdEndIndex = 0;
-                // if cmd starts with quotes, extract the cmd with quotes.
-                if (applicationPathAndArguments.indexOf("\"") == 0) {
-                    cmdEndIndex = applicationPathAndArguments.substring(1).indexOf('"');
-                    if (cmdEndIndex > -1) {
-                        cmdEndIndex = cmdEndIndex + 2;
-                    }
-                } else {
-                    // a cmd without quotes is split by a space or can be without arguments.
-                    cmdEndIndex = applicationPathAndArguments.indexOf(" ");
-                }
-                //if the cmdEndIndex < 0, we did not find a space or endQuote, so treat the whole string as command.
-                if (cmdEndIndex < 0) {
-                    logger.warning("Only command, no arguments: " + applicationPathAndArguments);
-                    cmdEndIndex=applicationPathAndArguments.length();                 
-                }
-                // cmdEndIndex++;
-                String cmdString = applicationPathAndArguments.substring(0, cmdEndIndex);
-                // The command can be an absolute path, so ensure that it has proper system separators (\ or \\ or /)
-                command.add(FilenameUtils.separatorsToSystem(cmdString));
-
-                String arguments = "";
-                try {
-                    arguments = applicationPathAndArguments.substring(cmdEndIndex + 1);
-                } catch (IndexOutOfBoundsException e) {
-                    logger.fine("No cmd arguments found.");
-                }
+                arguments = applicationPathAndArguments.substring(cmdEndIndex + 1);
+            } catch (IndexOutOfBoundsException e) {
+                logger.fine("No cmd arguments found.");
+            }
 
-                Matcher m = Pattern.compile("([^\"]\\S*|\".+?\")\\s*").matcher(arguments);
-                while (m.find()) {
-                    String token = m.group(1).replace("\"", "");
-                    command.add(token);
-                    logger.fine("adding argument: " + token);
-                }
+            Matcher m = Pattern.compile("([^\"]\\S*|\".+?\")\\s*").matcher(arguments);
```

Comment:
Please provide link to pattern tester and examples of tested inputs.

Is the following argument matched correctly?
```
-e "echo "This is a test" > output.txt"
```

Reply:
now using this refined regex: `(([^\s"]+)|(".+?"))\s*`
see comment in code for explanation.
Tests done on http://www.regexe.com/:
Input:
```
-e "echo "This is a test" > output.txt"
xterm -e "sudo sh test.sh"
cmd /c ipconfig
```
groups output:
```
-e
"echo "
This
is
a
test
" > output.txt"
xterm
-e
"sudo sh test.sh"
cmd
/c
ipconfig
```
all as expected.